Structure.Rmd

```diff
@@ -235,20 +243,52 @@ lapply(.libPaths(), dir)
 
-```{r, error = TRUE}
-library(blah)
-# or:
-require(blah)
+lapply(.libPaths(), list.dirs, recursive = FALSE, full.names = FALSE)
+#> [[1]]
+#>    [1] "abc"                  "abc.data"             "abind"                
+#>  ...
+#> [1033] "Zelig"                "zip"                  "zoo"                 
+#> 
+#> [[2]]
+#>  [1] "base"         "boot"         "class"        "cluster"     
+#>  [5] "codetools"    "compiler"     "datasets"     "foreign"     
+#>  [9] "graphics"     "grDevices"    "grid"         "KernSmooth"  
+#> [13] "lattice"      "MASS"         "Matrix"       "methods"     
+#> [17] "mgcv"         "nlme"         "nnet"         "parallel"    
+#> [21] "rpart"        "spatial"      "splines"      "stats"       
+#> [25] "stats4"       "survival"     "tcltk"        "tools"       
+#> [29] "translations" "utils"
 ```
 
-The main difference between `library()` and `require()` is what happens if a package isn't found. While `library()` throws an error, `require()` prints a warning message and returns `FALSE`. In practice, this distinction isn't important because when building a package you should __NEVER__ use either inside a package. See [package dependencies](#dependencies) for what you should do instead.
+In both cases we see two active libraries, consulted in this order:
 
-When you start learning R, it's easy to get confused between libraries and packages because you use `library()` function to load a _package_. However, the distinction between the two is important and useful. For example, one important application is packrat, which automates the process of managing project-specific libraries. With packrat, when you upgrade a package in one project, it only affects that project, not every project on your computer. This is useful because it allows you to play around with cutting-edge packages without affecting other projects' use of older, more reliable packages. This is also useful when you're both developing and using a package.
-t
+  1. A user library
+  2. A system-level or global library
+
+This setup is typical on Windows, but is something you usually need to opt into on macOS, for example, by specifying `R_LIBS_USER` in `.Renviron`. With this setup, add-on packages installed from CRAN (or elsewhere) or under local development are kept in the user library. Above, the macOS system is used as a primary development machine and has many packages here (~1000), whereas the Windows system is only used occasionally and is much more spartan. The core set of base and recommended packages that ship with R live in the system-level library and are the same on macOS and Windows. This separation appeals to many developers and makes it easy to, for example, clean out your add-on packages without disturbing your base R installation. If you're on macOS and only see one library, there is no urgent need to change anything, but it's something to consider next time you upgrade R.
```

Comment:
This seems like a bit of tease: what exactly should you do when you next update R?